File: MyLanguageWeb/src/main/resources/static/js/classes/dto/entity/workout/types/favourite/favourite_workout_type_response_dto.js
import {
    WorkoutTypeResponseDTO
} from "../../../workout_type/response/workout_type_response_dto.js";

export class FavouriteWorkoutTypeResponseDTO {
    #numberOfWorkouts;
    #workoutType;

    constructor(json) {
        if (json) {
            this.#numberOfWorkouts = json["number_of_workouts"];

            let workoutType = json["workout_type"];
            if (workoutType) {
                this.#workoutType = new WorkoutTypeResponseDTO(workoutType);
            }
        }
    }

    getNumberOfWorkouts() {
        return this.#numberOfWorkouts;
    }

    getWorkoutType() {
        return this.#workoutType;
    }
}

File: MyLanguageWeb/src/main/java/ru/herooo/mylanguageweb/dto/entity/workout/types/favourite/workout_type/FavouriteWorkoutTypeResponseDTO.java
package ru.herooo.mylanguageweb.dto.entity.workout.types.favourite.workout_type;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
import com.fasterxml.jackson.databind.annotation.JsonSerialize;
import ru.herooo.mylanguageweb.json.l.LongDeserializer;
import ru.herooo.mylanguageweb.json.l.LongSerializer;
import ru.herooo.mylanguageweb.dto.entity.workouttype.response.WorkoutTypeResponseDTO;

public class FavouriteWorkoutTypeResponseDTO {
    @JsonProperty("workout_type")
    private WorkoutTypeResponseDTO workoutType;

    @JsonSerialize(using = LongSerializer.class)
    @JsonDeserialize(using = LongDeserializer.class)
    @JsonProperty("number_of_workouts")
    private long numberOfWorkouts;

    public WorkoutTypeResponseDTO getWorkoutType() {
        return workoutType;
    }

    public void setWorkoutType(WorkoutTypeResponseDTO workoutType) {
        this.workoutType = workoutType;
    }

    public long getNumberOfWorkouts() {
        return numberOfWorkouts;
    }

    public void setNumberOfWorkouts(long numberOfWorkouts) {
        this.numberOfWorkouts = numberOfWorkouts;
    }
}
